aten/src/ATen/native/mps/operations/Blas.mm

```diff
@@ -51,13 +51,35 @@ Tensor dot_mps(
           MPSGraphTensor *selfTensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, self);
           MPSGraphTensor *otherTensor =  mps::mpsGraphRankedPlaceHolder(mpsGraph, other);
 
-          MPSGraphTensor *dot = [mpsGraph multiplicationWithPrimaryTensor: selfTensor
-                                                          secondaryTensor: otherTensor
+          MPSGraphTensor *castSelf = nil;
+          MPSGraphTensor *castOther = nil;
+
+          if(self.scalar_type() == ScalarType::Short || self.scalar_type() == ScalarType::Byte) {
+            castSelf = [mpsGraph castTensor:selfTensor
+                                     toType:MPSDataTypeInt32
+                                       name:@"castSelfTensor"];
+            castOther = [mpsGraph castTensor:otherTensor
```

Comment:
Why there's no separate condition before casting otherTensor? I mean, why do you reuse the type checking condition of selfTensor for otherTensor too?

Reply:
Hmm, I guess I assume for now that self and other have the same type, as is expected for dot operation... we should be doing more checking in case the inputs are not the same type, but I think that should be done in a separate PR